### PR TITLE
Headline | Docs | Testing and Documentation for Headline Component

### PR DESCRIPTION
**Pull request by TashaBasalt**
## Jira

http://vjira2:8080/browse/BDS-986

## Summary

Adding Jest Snapshot tests of headline component and a TESTING.md file.

## Details

Added tests to cover various combinations of font sizes and weights, but I'm not sure that amount of specificity is required. There is also a 'quoted' example of text in the Bolt Pattern Lab, but the sample text isn't actually quoted in any way. Is that an error?

## How to test

yarn test:js

### CODE REVIEW

**Comment by margoromo** [comment]
Hey @mikemai2awesome  - can you review this? Some questions in the PR.

File: packages/components/bolt-headline/TESTING.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# List component testing steps
```
Comment (joekarasek) [inline packages/components/bolt-headline/TESTING.md]: Typo.

File: packages/components/bolt-headline/TESTING.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+## List component render as expected (functionally and visually)
+
+The server-side pre-rendered `bolt-headline` component should look almost identical to the client-side rendered version. To verify:
+
+1. Disable javascript and view the [Headline Demo page](https://boltdesignsystem.com/pattern-lab/patterns/02-components-headline-05-headline/02-components-headline-05-headline.html).
+2. Then, enable javascript and watch as the List re-renders on the client-side.
+3. The layout should not shift, e.g. the spacing in between each list item should be identical before and after the web component re-renders.
```
Comment (joekarasek) [inline packages/components/bolt-headline/TESTING.md]: Same with the code-snippet. Go ahead an run this test yourself and confirm its behavior. Probably no list item spacing issue, but we do want to make sure the font weight, family, line-height, etc dont change or shift.

File: packages/components/bolt-headline/TESTING.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+## Scenario: tag variations
+
+1. Given I am viewing the [tag variations page](https://boltdesignsystem.com/pattern-lab/patterns/02-components-headline-10-headline-tag-variations/02-components-headline-10-headline-tag-variations.html).
+2. A List of header tags(H1-H6), a paragragh tag(p), and a span tag.
+3. Below each tag should be a text sample that shows the font size each tag would produce.
+
+## Scenario: alignment variations
+
+1. Given I am viewing the [alignment variations page](https://boltdesignsystem.com/pattern-lab/patterns/02-components-headline-15-headline-align-variations/02-components-headline-15-headline-align-variations.html).
+2. There should be three Align headings.
```
Comment (joekarasek) [inline packages/components/bolt-headline/TESTING.md]: List the three here. A little overkill, but being overly verbose will help the QA team.

File: packages/components/bolt-headline/TESTING.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+3. Beneath each should be text samples aligned at the far left, the center, and the far right of the page.
+
+## Scenario: style and weight variations
+
+1. Given I am viewing the [style and weight variations page](https://boltdesignsystem.com/pattern-lab/patterns/02-components-headline-20-headline-style-and-weight-variations/02-components-headline-20-headline-style-and-weight-variations.html).
+2. A list of text samples in various combination of font size, font weight, and font styles.
```
Comment (joekarasek) [inline packages/components/bolt-headline/TESTING.md]: @mikemai2awesome We had some questions about this. It seems in the example that there are several places where the sample text says something like "This is a headline thats bold and italic", but its not bold or italic (guessing that the given combo is not something Bolt supports in those cases). Any suggestions around writing instructions for this without generating a whole ton of false positives from the QA team?

File: packages/components/bolt-headline/TESTING.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+## Scenario: icon variations
+
+1. Given I am viewing the [icon variations page](https://boltdesignsystem.com/pattern-lab/patterns/02-components-headline-25-headline-icon-variations/02-components-headline-25-headline-icon-variations.html).
+2. A list of headlines with three positioning options(before, after, and default), and a headline exhibiting the option to choose various different icons.
```
Comment (joekarasek) [inline packages/components/bolt-headline/TESTING.md]: Just a little more verbose here. Something along the lines of "an icon will appear before or after"

File: packages/components/bolt-headline/__tests__/headline.js [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+          },
+        );
+        expect(results.ok).toBe(true);
+        expect(results.html).toMatchSnapshot();
+      });
+    });
+  });
+
+  size.enum.forEach(async sizeChoice => {
+    const fontStyle = ['normal', 'italic'];
```
Comment (joekarasek) [inline packages/components/bolt-headline/__tests__/headline.js]: Let's lift these out of the schema file incase another style is added in the future.

File: packages/components/bolt-headline/__tests__/headline.js [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+        align: alignmentChoice,
+        text: 'Some text',
+      });
+      expect(results.ok).toBe(true);
+      expect(results.html).toMatchSnapshot();
+    });
+  });
+
+  size.enum.forEach(async sizeChoice => {
+    const fontWeight = ['bold', 'regular', 'semibold'];
```
Comment (joekarasek) [inline packages/components/bolt-headline/__tests__/headline.js]: Lift from schema

File: packages/components/bolt-headline/__tests__/headline.js [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+  test('headline with associated link', async () => {
+    const results = await render('@bolt-components-headline/headline.twig', {
+      url: 'www.testurl.com',
+      text: 'this is a headline',
+    });
+    expect(results.ok).toBe(true);
+    expect(results.html).toMatchSnapshot();
+  });
+
+  test('headline with icon', async () => {
```
Comment (joekarasek) [inline packages/components/bolt-headline/__tests__/headline.js]: @TashaBasalt Looks like there is an option to place an icon before or after. See below. Lets add these as tests.
```<h3>Headline w/ Icon Position <code>after</code></h3>
{% include "@bolt-components-headline/headline.twig" with {
  text: "What we do at Pega is brilliant!",
  icon: {
    name: "chevron-right",
    position: "after"
  }
} only %}
<br><br>```
@mikemai2awesome Do you feel that its necessary to add test coverage for the deprecated versions of the this (position: left and position: right)?

File: packages/components/bolt-headline/TESTING.md (hunk 2 above)
Comment (TashaBasalt) [reply packages/components/bolt-headline/TESTING.md]: Tested with no issue.

File: packages/components/bolt-headline/TESTING.md (hunk 4 above)
Comment (mikemai2awesome) [reply packages/components/bolt-headline/TESTING.md]: @joekarasek 

That page is so wrong. I would skip it. We want `bolt-text` to replace headline eventually, which already has way better documentation: https://master.boltdesignsystem.com/pattern-lab/?p=components-text-typographic-recipes

File: packages/components/bolt-headline/TESTING.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+## Scenario: link variations
+
+1. Given I am viewing the [link variations page](https://boltdesignsystem.com/pattern-lab/patterns/02-components-headline-30-headline-link-variations/02-components-headline-30-headline-link-variations.html).
+2. Three links that show a default icon with an arrow, a defined icon with a magnifying glass, or no icon.
+
+## Scenario: quoted variations
+
+1. Given I am viewing the [quoted variations page](https://boltdesignsystem.com/pattern-lab/patterns/02-components-headline-35-headline-quoted-variation/02-components-headline-35-headline-quoted-variation.html).
+2. A single headline.
```
[the remark this answers is not in the document]
Comment (TashaBasalt) [reply packages/components/bolt-headline/TESTING.md]: Visually the text isn't quoted in the example page, looking at the snapshots as well it doesn't seem that the `quoted` option is working properly.

[the remark this answers is not in the document]
Comment (joekarasek) [reply packages/components/bolt-headline/TESTING.md]: @margoromo @mikemai2awesome Looks like the headline element is not properly rendering quotes. I updated the TESTING.md text to reflect that there should be quotes visible. Do we want to leave this ticket open until it's fix, or can we move forward with closing it?

**Comment by sghoweri** [comment]
@joekarasek @TashaBasalt just checking in -- what's the status on this one?

**Comment by TashaBasalt** [comment]
@sghoweri We're still waiting on confirmation for whether or not to have the TESTING.md describe a successful function that isn't actually functional.

**Comment by joekarasek** [comment]
@sghoweri @mikemai2awesome This is now ready for final review/merge.
1) Re: quoted variation. We added "text should be quoted" to the TESTING.md even though that feature does not seem to be working.
2) Looks like the build failed on a test unrelated to this work "auto-link replacement of urls on bolt-button". Gonna do a fresh merge of master and a new deploy to see if it resolves the issue. I'm assuming its either an out of date dependency or a one-off false positive.

**Comment by sghoweri** [comment]
Thanks guys!!

**Pull request merged**